db/memorystore.go

```diff
@@ -42,6 +42,16 @@ func (m *MemoryStore) GetAccountByID(id string) *core.Account {
 	return m.accountsByID[id]
 }
 
+func (m *MemoryStore) UpdateAccountByID(id string, acct *core.Account) error {
+	m.RLock()
```

Comment:
I think this needs to be a `m.Lock()` with a `defer r.Unlock()` because you are changing `accountsByID`.

Reply:
Nice catch - I missed that this was a RWLock!